### PR TITLE
Cleaning up headers and examples

### PR DESCRIPTION
**Pull request by JesperIRL**
* Make all headers follow the same style
* Make code examples use ~~~ instead of four space indentation
* Fix email adresses to be actual adresses
* Add link to jtreg FAQ
* Add comment about the name space for noreg-labels

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Stuart Marks](https://openjdk.org/census#smarks) (@stuart-marks - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/guide pull/90/head:pull/90` \
`$ git checkout pull/90`

Update a local copy of the PR: \
`$ git checkout pull/90` \
`$ git pull https://git.openjdk.org/guide pull/90/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 90`

View PR using the GUI difftool: \
`$ git pr show -t 90`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/guide/pull/90.diff">https://git.openjdk.org/guide/pull/90.diff</a>

</details>

### CODE REVIEW

**Review by irisclark — APPROVED** [review]

**Review by stuart-marks — APPROVED** [review]

File: src/guide/mailing-lists.md [hunk 1]
```diff
@@ -23,13 +23,13 @@ There are a few different types of lists. The list name has two parts to explain
 > :    Technical discussions around the usage of the project artifacts.
 
 > `-discuss`
-> :    General discussions around the project. The special case `discuss(at)openjdk.org` is used for general discussions around OpenJDK. Discussions around new project proposals usually happens here.
+> :    General discussions around the project. The special case `discuss@openjdk.org` is used for general discussions around OpenJDK. Discussions around new project proposals usually happens here.
```
Comment (aivanov-jdk) [inline src/guide/mailing-lists.md:26]:
```suggestion
> :    General discussions around the project. The special case `discuss@openjdk.org` is used for general discussions around OpenJDK. Discussions around new project proposals usually happen here.
```

File: src/guide/testing-the-jdk.md [hunk 2]
```diff
@@ -30,43 +30,53 @@ In-depth documentation about the jtreg framework is found here: [jtreg harness](
...
 
 Or you can specify a number of requirements that must be fulfilled for jtreg to execute the test.
 
-    @requires docker.support
-    @requires os.family != ”windows”
-    @requires os.maxMemory > 3G
-    @requires os.arch=="x86_64" | os.arch=="amd64"
+~~~
+@requires docker.support
+@requires os.family != ”windows”
```
Comment (aivanov-jdk) [inline src/guide/testing-the-jdk.md:61]:
```suggestion
@requires os.family != "windows"
```
The quotes should be plain for easier copy-and-paste.

File: src/guide/working-with-the-legacy-mercurial-servers.md [hunk 3]
```diff
@@ -221,7 +221,7 @@ In order to push changesets into the parent repository, some additional configur
 
 ### Get your SSH key installed
```
Comment (aivanov-jdk) [inline src/guide/working-with-the-legacy-mercurial-servers.md:222]: Should this header use title capitalisation as other headers, at least in this review?

Comment (JesperIRL) [reply src/guide/working-with-the-legacy-mercurial-servers.md:222]: No, currently only first level headers are capitalized in this document.

**Comment by JesperIRL** [comment]
@aivanov-jdk Sorry, I didn't see your comments before integrating. I tried to remove the /integrate comment, but the bots were unusually fast today so they saw it anyway. I'll include your changes in my next patch.

File: src/guide/testing-the-jdk.md (hunk 2 above)
Comment (JesperIRL) [reply src/guide/testing-the-jdk.md:61]: Good catch!

File: src/guide/mailing-lists.md (hunk 1 above)
Comment (JesperIRL) [reply src/guide/mailing-lists.md]: Fixed.

**Comment by aivanov-jdk** [comment]
> @aivanov-jdk Sorry, I didn't see your comments before integrating. I tried to remove the /integrate comment, but the bots were unusually fast today so they saw it anyway. I'll include your changes in my next patch.

Sounds good, thanks.

**Pull request closed without merging**